### PR TITLE
doc: Configure Postgres CDC on Cloud services

### PR DESCRIPTION
**Pull request by msudberg**
Added instructions on configuring Amazon RDS and Cloud SQL for Postgres CDC.

### Motivation
 
  * This PR adds a known-desirable feature. [#7235 ]
 

### Tips for reviewer

Google has instructions about [configuring pglogical ](https://cloud.google.com/sql/docs/postgres/replication/configure-logical-replication) -- do we need to do that for the upstream database and/or for Materialize?

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR adds a release note for any
  [user-facing behavior changes](/doc/user/content/release-notes.md#What-changes-require-a-release-note).

### CODE REVIEW

**Comment by msudberg** [comment]
> Should we also mention somewhere that our Postgres CDC works with AWS Aurora?

Yes, definitely. Are the configuration steps the same as for RDS? The [AWS doc](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/AuroraPostgreSQL.Replication.Logical.html) mentions several additional steps, but then so did Cloud SQL.

**Review by alex-hunt-materialize — COMMENTED** [review]
I have no idea what the requirements for CDC are in cloud services, but the parts pertaining to Materialize Cloud look good.

File: doc/user/content/guides/postgres-cloud.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+For more information, see the [AWS Aurora](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/AuroraPostgreSQL.Replication.Logical.html#AuroraPostgreSQL.Replication.Logical.Configure) documentation.
+
+
+## Cloud SQL
+
+As a user with the `cloudsqlsuperuser` role, make these changes to the upstream database:
+
+1. The Materialize replica will need access to connect to the upstream database. In the Google Cloud Console, enable access on the upstream database for the Materialize replica's IP address.
+
+      If you are using Materialize Cloud, you will need to make the RDS instance publically accessible; Materialize Cloud instances do not have static IP addresses.
```
Comment (petrosagg) [inline doc/user/content/guides/postgres-cloud.md:103]: s/RDS/Cloud SQL

File: doc/user/content/guides/postgres-cloud.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    The `mz_source` publication will contain the set of change events generated from the specified tables, and will later be used to ingest the replication stream. We strongly recommend that you **limit** publications only to the tables you need.
+
+For more information, see the [AWS Aurora](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/AuroraPostgreSQL.Replication.Logical.html#AuroraPostgreSQL.Replication.Logical.Configure) documentation.
+
+
+## Cloud SQL
+
+As a user with the `cloudsqlsuperuser` role, make these changes to the upstream database:
+
+1. The Materialize replica will need access to connect to the upstream database. In the Google Cloud Console, enable access on the upstream database for the Materialize replica's IP address.
```
Comment (petrosagg) [inline doc/user/content/guides/postgres-cloud.md:101]: Aren't we missing a step here to enable logical replication?

**Review by petrosagg — APPROVED** [review]
Looks good, I left a couple of comments that can be fixed async

File: doc/user/content/guides/postgres-cloud.md (hunk 2 above)
Comment (msudberg) [reply doc/user/content/guides/postgres-cloud.md:102]: Added.

**Pull request merged**